pilot/pkg/model/telemetry.go

```diff
@@ -434,6 +461,10 @@ func mergeLogs(logs []*tpb.AccessLogging, mesh *meshconfig.MeshConfig) (sets.Set
 		if m.Filter != nil {
 			loggingFilter = m.Filter
 		}
+
+		if m.Match != nil {
+			loggingSelector = m.Match
```

Comment:
I donot think the merging is right, now this is merging all logging that could be applied to a proxy, but does not care whether the match selectors are same or not.

Reply:
Agreed. I think we need something that is aware of selector logic in merging.

Even with a single provider, the following scenario:

mesh-wide: server-side logging enabled, client-side logging disabled
namespace-wide: server-side logging enabled with filter `foo`
workload-specific: client-side logging enabled with filter `bar`

should result in:

server-side logging with filter `foo` and client-side logging with filter `bar`.

This would seem to set to result in only client-side logging with filter `bar`.

I _think_ the `mergeLogs` needs more complex updating as a result.

Reply:
so you means we need merge `selector` but not merge `filter`?

Reply:
updated, please take a review again @douglas-reid